worker/src/ocall_bridge/attestation_ocall_impl.rs

```diff
@@ -0,0 +1,163 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+    Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use crate::ocall_bridge::bridge_api::RemoteAttestationOCall;
+use log::*;
+use sgx_types::*;
+use std::net::{SocketAddr, TcpStream};
+use std::os::unix::io::IntoRawFd;
+use std::ptr;
+
+pub struct RemoteAttestationOCallImpl {
+    // TODO as a member here we need the e-call API trait, so we can use it instead of making the e-call directly
+}
+
+impl RemoteAttestationOCall for RemoteAttestationOCallImpl {
+    fn init_quote(&self) -> (sgx_status_t, sgx_target_info_t, sgx_epid_group_id_t) {
+        // TODO this translation to unsafe C-API should be moved to the EnclaveApi / ECall API
+        let mut ti: sgx_target_info_t = sgx_target_info_t::default();
+        let mut eg: sgx_epid_group_id_t = sgx_epid_group_id_t::default();
+
+        unsafe {
+            let ret_status = sgx_init_quote(
+                &mut ti as *mut sgx_target_info_t,
+                &mut eg as *mut sgx_epid_group_id_t,
+            );
+            (ret_status, ti, eg)
+        }
+    }
```

Comment:
3 of the 4 methods for remote attestation basically just call back right into the enclave, using an e-call. Once we have the e-call API for this, these implementations should be moved to that API, making the implementation here very simple.